### PR TITLE
UCT/TCP: Keep all receive sockets in a hash table.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/uct/tcp/tcp.h [hunk 1]
```diff
@@ -13,50 +13,63 @@
...
+
+
 /**
  * TCP endpoint
  */
 typedef struct uct_tcp_ep {
-    uct_base_ep_t       super;
-    int                 fd;
+    uct_base_ep_t                 super;
+    int                           fd;        /* Socked filed descriptor */
```
Comment (miked-mellanox) [inline src/uct/tcp/tcp.h]: typo: socked, filed.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3179/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/705/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3180/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/706/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2653/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2654/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@brminich @alex-mikheev pls review

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2664/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2667/ for details (Mellanox internal link).

File: src/uct/tcp/tcp_ep.c [hunk 2]
```diff
@@ -20,22 +20,36 @@ static UCS_CLASS_INIT_FUNC(uct_tcp_ep_t, uct_iface_t *tl_iface,
...
+        goto err;
+    }
+
+    status = uct_tcp_iface_set_sockopt(iface, self->fd);
+    if (status != UCS_OK) {
+        goto err_close;
     }
 
     memset(&dest_addr, 0, sizeof(dest_addr));
+    dest_addr.sin_addr.s_addr = INADDR_ANY;
```
Comment (brminich) [inline src/uct/tcp/tcp_ep.c]: Why is this needed?

File: src/uct/tcp/tcp_iface.c [hunk 3]
```diff
@@ -283,6 +275,7 @@ static UCS_CLASS_DEFINE_NEW_FUNC(uct_tcp_iface_t, uct_iface_t, uct_md_h,
                                  uct_worker_h, const uct_iface_params_t*,
                                  const uct_iface_config_t*);
 
+
```
Comment (brminich) [inline src/uct/tcp/tcp_iface.c]: Is there a  need for extra line here? Single line is used in the rest of the fille.

**Comment by brminich** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3192/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/721/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2671/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2673/ for details (Mellanox internal link).
